src/blog/oilpan-library.md

```diff
@@ -0,0 +1,51 @@
+
+Early 2020, we started a journey in carving out Oilpan from Blink and encapsulating it into a library. We decided to host the code in V8, reuse abstractions where possible, and do some spring cleaning on the garbage collection interface. In addition to fixing all of the aforementioned issues, [a library](https://docs.google.com/document/d/1ylZ25WF82emOwmi_Pg-uU6BI1A-mIbX_MG9V87OFRD8/) would also enable other projects to make use of garbage-collected C++. We launched the library in V8 v9.4 and enabled it in Blink starting in Chromium M94.
+
+## What’s in the box?
+
+Similar to the rest of V8, Oilpan now provides a [stable API](https://chromium.googlesource.com/v8/v8.git/+/HEAD/include/cppgc/) and embedders may rely on the regular [V8 conventions](https://v8.dev/docs/api). For example, this means that APIs are properly documented (see [GarbageCollected](https://chromium.googlesource.com/v8/v8.git/+/main/include/cppgc/garbage-collected.h#17)) and will go through a deprecation period in case they are subject to being removed or changed.
+
+The core of Oilpan is available as a stand-alone C++ garbage collector in the cppgc namespace. The setup also allows for reusing an existing V8 platform to create a heap for managed C++ objects. Garbage collections can be configured to run automatically, integrating in the task infrastructure or can be triggered explicitly considering the native stack as well. The idea is to allow embedders that just want managed C++ objects to avoid dealing with V8 as a whole, see this [hello world program](https://chromium.googlesource.com/v8/v8.git/+/main/samples/cppgc/hello-world.cc) as an example. Conveniently, tests for the core of Oilpan use this setup which means that it’s a matter of seconds to build and run a specific garbage collection test. As of today, [>400 of such unit tests](https://source.chromium.org/chromium/chromium/src/+/main:v8/test/unittests/heap/cppgc/) for the core of Oilpan exist. The setup also serves as a playground to experiment and try out new things and may be used to validate assumptions around raw performance. An embedder of this configuration is PDFium which uses Oilpan’s stand-alone version for [securing XFA](https://groups.google.com/a/chromium.org/g/chromium-dev/c/RAqBXZWsADo/m/9NH0uGqCAAAJ?utm_medium=email&utm_source=footer) which allows for more dynamic PDF content.
```

Comment:
```suggestion
The core of Oilpan is available as a stand-alone C++ garbage collector in the `cppgc` namespace. The setup also allows for reusing an existing V8 platform to create a heap for managed C++ objects. Garbage collections can be configured to run automatically, integrating in the task infrastructure or can be triggered explicitly considering the native stack as well. The idea is to allow embedders that just want managed C++ objects to avoid dealing with V8 as a whole, see this [hello world program](https://chromium.googlesource.com/v8/v8.git/+/main/samples/cppgc/hello-world.cc) as an example. Conveniently, tests for the core of Oilpan use this setup which means that it’s a matter of seconds to build and run a specific garbage collection test. As of today, [>400 of such unit tests](https://source.chromium.org/chromium/chromium/src/+/main:v8/test/unittests/heap/cppgc/) for the core of Oilpan exist. The setup also serves as a playground to experiment and try out new things and may be used to validate assumptions around raw performance. An embedder of this configuration is PDFium which uses Oilpan’s stand-alone version for [securing XFA](https://groups.google.com/a/chromium.org/g/chromium-dev/c/RAqBXZWsADo/m/9NH0uGqCAAAJ?utm_medium=email&utm_source=footer) which allows for more dynamic PDF content.
```

This one caught my eye, but if there are other things that relate to code, it's best to wrap them into backticks too.

Reply:
I'd also split this paragraph into smaller pieces as well, e.g. the test one can be neatly split out into its own part:

```suggestion
The core of Oilpan is available as a stand-alone C++ garbage collector in the `cppgc` namespace. The setup also allows for reusing an existing V8 platform to create a heap for managed C++ objects. Garbage collections can be configured to run automatically, integrating in the task infrastructure or can be triggered explicitly considering the native stack as well. The idea is to allow embedders that just want managed C++ objects to avoid dealing with V8 as a whole, see this [hello world program](https://chromium.googlesource.com/v8/v8.git/+/main/samples/cppgc/hello-world.cc) as an example.

Conveniently, tests for the core of Oilpan use this setup which means that it’s a matter of seconds to build and run a specific garbage collection test. As of today, [>400 of such unit tests](https://source.chromium.org/chromium/chromium/src/+/main:v8/test/unittests/heap/cppgc/) for the core of Oilpan exist.

The setup also serves as a playground to experiment and try out new things and may be used to validate assumptions around raw performance. An embedder of this configuration is PDFium which uses Oilpan’s stand-alone version for [securing XFA](https://groups.google.com/a/chromium.org/g/chromium-dev/c/RAqBXZWsADo/m/9NH0uGqCAAAJ?utm_medium=email&utm_source=footer) which allows for more dynamic PDF content.
```

Reply:
Re: code references - I see a few more namespaces mentioned throughout the post; it's best to add backticks in those so that they remain links but also clearly are marked as code.

Reply:
Looks like this was accidentally marked as resolved - the other namespaces still need to be wrapped too as per last comment.

Reply:
Yeah, indeed.

- I now added more ticks around C++ names.
- I split the paragraph but did so slightly differently.

Reply:
Thanks, looks great!